### PR TITLE
Apply archboard review recommendations

### PR DESCRIPTION
**Pull request by lzchen**
Moves certain paths to protected to prevent from being seen by api review, as they are not expected to be called by the user.
Add docstring for `configure_azure_monitor` api.
Pin opentelemetry api/sdk versions

### CODE REVIEW

File: azure-monitor-opentelemetry-distro/azure/monitor/opentelemetry/distro/version.py [hunk 1]
```diff
@@ -1,7 +0,0 @@
-# -------------------------------------------------------------------------
-# Copyright (c) Microsoft Corporation. All rights reserved.
-# Licensed under the MIT License. See License in the project root for
-# license information.
-# --------------------------------------------------------------------------
-
-VERSION = "1.0.0b8"
```
Comment (jeremydvoss) [inline azure-monitor-opentelemetry-distro/azure/monitor/opentelemetry/distro/version.py:7]: Where are we setting the version instead?

Comment (jeremydvoss) [reply azure-monitor-opentelemetry-distro/azure/monitor/opentelemetry/distro/version.py:7]: Was this supposed to be a _version rename?

Comment (lzchen) [reply azure-monitor-opentelemetry-distro/azure/monitor/opentelemetry/distro/version.py:7]: https://github.com/microsoft/ApplicationInsights-Python/blob/main/azure-monitor-opentelemetry-distro/setup.py#L40

Comment (jeremydvoss) [reply azure-monitor-opentelemetry-distro/azure/monitor/opentelemetry/distro/version.py:7]: Got it. We never deleted the old one.

**Review by jeremydvoss — APPROVED** [review]

**Pull request merged**